tests/integration/test_client.py

```diff
@@ -88,3 +91,35 @@ async def test_command_validation_failure(harness):
             'response': [False, '--pre=all must be used alone'],
         }
     ]
+
+
+@pytest.mark.parametrize(
+    'sock_response, expected',
+    [
+        pytest.param({'error': 'message'}, r"^message$", id="basic"),
+        pytest.param(
+            {'foo': 1},
+            r"^Received invalid response for Cylc 8\.[\w.]+: \{'foo': 1[^}]*\}$",
+            id="no-err-field",
+        ),
+        pytest.param(
+            {'cylc_version': '8.x.y'},
+            r"^Received invalid.+\n\(Workflow is running in Cylc 8.x.y\)$",
+            id="no-err-field-with-version",
+        ),
+    ],
+)
+async def test_async_request_err(
+    one, start, monkeypatch: pytest.MonkeyPatch, sock_response, expected
+):
+    async def mock_recv():
+        return json.dumps(sock_response).encode()
+
+    async with start(one):
+        client = WorkflowRuntimeClient(one.workflow)
+        with monkeypatch.context() as mp:
+            mp.setattr(client, 'socket', Mock(recv=mock_recv))
+            mp.setattr(client, 'poller', Mock())
```

Comment:
For future reference, I originally tried only mocking the `poll` method of `client.poller`, i.e.
```python
mp.setattr(client.poller, 'poll', Mock())
```

but this resulted in pytest hanging near the end of running all integration tests; some strange interaction due to the ZMQ poller using threading or something, dunno.